app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala

```diff
@@ -86,14 +89,13 @@ class BitcoinSServerMain(override val serverArgParser: ServerArgParser)(implicit
   override def stop(): Future[Unit] = {
     logger.error(s"Exiting process")
     for {
-      _ <- walletConf.stop()
-      _ <- nodeConf.stop()
-      _ <- chainConf.stop()
-      _ <- torConf.stop()
+      _ <- conf.stop()
+      _ <- serverBindingsOpt match {
+        case Some(bindings) => bindings.stop()
+        case None           => Future.unit
+      }
       _ = logger.info(s"Stopped ${nodeConf.nodeType.shortName} node")
-      _ <- system.terminate()
```

Comment:
what's the reason not to terminate the `system` here?

Reply:
In the case of using `BitcoinSServerMain` test case fixtures, this terminates the `ActorSystem` that is shared throughout the test suite. This causes tests to hang inside of `WebsocketTest` if we don't remove this.

![Screenshot from 2021-12-23 13-49-26](https://user-images.githubusercontent.com/3514957/147285879-ab196fae-963f-4741-baa5-71d728dfe979.png)

Reply:
I see, makes sense, perhaps this should be made conditional on the `isTest` environment variable thing? But maybe not, you'd know better than me

Reply:
I don't think its worth the hassle